### ISSUE TITLE
Latest and Stable versions failing due to invalid config options 

### ISSUE THREAD

**Issue by discovery-NukulSharma** [aws/aws-for-fluent-bit#371]
We are noticing stable version since yesterday are failing, seems previous version is working fine , if we just use specific tagof previous version. 

Failed

- public.ecr.aws/aws-observability/aws-for-fluent-bit:stable
- public.ecr.aws/aws-observability/aws-for-fluent-bit:latest

Passed - older version

- public.ecr.aws/aws-observability/aws-for-fluent-bit:2.23.3

**Comment by JasonIamAUnixAdmin** [comment]
Same here. Using the provided params by AWS cause this to fail to start with a 255 error taking the rest of the task with it. When past default values causes crashes this should not be in "stable"

**Comment by zhonghui12** [comment]
@discovery-NukulSharma @JasonIamAUnixAdmin Thanks for report. Working on finding the root cause

**Comment by zhonghui12** [comment]
Also @discovery-NukulSharma @JasonIamAUnixAdmin , could you please explain more about what "failing" means? And could any of you confirm for me that if our dockerhub images have the same problem or the issue only happens with public ECR? https://hub.docker.com/r/amazon/aws-for-fluent-bit/tags

**Comment by JasonIamAUnixAdmin** [comment]
Failing: the log container exits with a 255 error code, since it is tagged as essential it takes the whole task out.

Pinning to the old "stable" image of 2.23.3 fixes the issue, using the current stable has the same issue.
We use this to send logs to DataDog, error seems related to https://github.com/aws/aws-for-fluent-bit/issues/348 

seems that our requirement of passing in `DD_LOGS_CONFIG_LOGS_DD_URL` which is a needed to change the endpoint used to ship logs might be the culprit

**Comment by JasonIamAUnixAdmin** [comment]
> 

correct. We changed to `:stable` rather than the default provided by the tools of `:latest` to get away from this kind of breaking change. 

And since we are messing with logging I understand why it is hard to get the "why" out out to show us but man is this painful to debug on the user end.

**Comment by zhonghui12** [comment]
@JasonIamAUnixAdmin  so from the https://github.com/aws/aws-for-fluent-bit/issues/348 we can see that from fluent bit 1.9, it starts to do some input validation. Can you please share your config so that we can figure out which field is not allowed?

**Comment by JasonIamAUnixAdmin** [comment]
Here is one task:
```
 "logConfiguration": {
        "logDriver": "awsfirelens",
        "secretOptions": [
          {
            "valueFrom": "arn:aws:secretsmanager:us-east-1:XXX:secret:YYY",
            "name": "apikey"
          }
        ],
        "options": {
          "provider": "ecs",
          "dd_service": "peach_scheduled",
          "dd_source": "peach",
          "DD_LOGS_CONFIG_LOGS_DD_URL": "tcp-encrypted-intake.logs.datadoghq.com:10516",
          "Name": "datadog"
        }
      },
  ```

And another:
```
     "logConfiguration": {
        "logDriver": "awsfirelens",
        "secretOptions": [
          {
            "valueFrom": "arn:aws:secretsmanager:us-east-1:XXX:secret:YYY",
            "name": "apikey"
          }
        ],
        "options": {
          "DD_DJANGO_DATABASE_SERVICE_NAME_PREFIX": "health-",
          "exclude-pattern": "/health_check/",
          "provider": "ecs",
          "dd_service": "health-staging-web",
          "dd_source": "health-staging",
          "DD_LOGS_CONFIG_LOGS_DD_URL": "tcp-encrypted-intake.logs.datadoghq.com:10516",
          "Name": "datadog"
        }
      },
```

**Comment by zhonghui12** [comment]
Thanks @JasonIamAUnixAdmin , so from the public repo: https://docs.fluentbit.io/manual/pipeline/outputs/datadog. not only `DD_LOGS_CONFIG_LOGS_DD_URL`, but also `DD_DJANGO_DATABASE_SERVICE_NAME_PREFIX` and `exclude-pattern` are invalid. Because the new validation in datadog: https://github.com/fluent/fluent-bit/commit/0e94b10bdbe1e36dc5b5a3600fd140789a29490c. Those should be removed to make fluent bit work from fluent bit 1.9

**Comment by JasonIamAUnixAdmin** [comment]
> Thanks @JasonIamAUnixAdmin , so from the public repo: https://docs.fluentbit.io/manual/pipeline/outputs/datadog. not only `DD_LOGS_CONFIG_LOGS_DD_URL`, but also `DD_DJANGO_DATABASE_SERVICE_NAME_PREFIX` and `exclude-pattern` are invalid. Because the new validation in datadog: [fluent/fluent-bit@0e94b10](https://github.com/fluent/fluent-bit/commit/0e94b10bdbe1e36dc5b5a3600fd140789a29490c). Those should be removed to make fluent bit work from fluent bit 1.9

All the options in use came from DD in the past. 

Looks like `HOST` replaces the logs URL
I don't see anything like the excludes pattern however

**Comment by zhonghui12** [comment]
@JasonIamAUnixAdmin  sorry I was wrong. So `exclude-pattern` and `include-pattern` are generated by FireLens and I've verified these two should still work: https://docs.aws.amazon.com/AmazonECS/latest/userguide/firelens-filtering-logs.html.

Thanks.

**Comment by zhonghui12** [comment]
@nakulpathak3 so right now we don't think the `stable` version is broken sorry. We can see the root cause here is wrong task definition because Fluent Bit 1.9 adds more input validation. Users should be able to control their definition configs and make it work. 

To be more specific, revert cannot resolve the issue because Fluent Bit will not revert their validation which means after Fluent Bit 1.9, the restriction will always there. I will recommend you to update your task definition config and if you have any problem, you can post it here and we are always willing you to help.

Thanks

**Comment by james-skinner-deltatre** [comment]
We also were hit by this. Logs confirm its a config validation issue:

> 2022-06-21 11:56:37AWS for Fluent Bit Container Image Version 2.25.1
...
2022-06-21 11:56:37[2022/06/21 09:56:37] [error] [config] datadog: unknown configuration property 'dd_env'. The following properties are allowed: compress, apikey, dd_service, dd_source, dd_tags, proxy, include_tag_key, tag_key, dd_message_key, provider, and json_date_key.
2022-06-21 11:56:37[2022/06/21 09:56:37] [ help] try the command: /fluent-bit/bin/fluent-bit -o datadog -h
2022-06-21 11:56:37[2022/06/21 09:56:37] [ info] [fluent bit] version=1.9.3, commit=eb4e2e770f, pid=1
2022-06-21 11:56:37[2022/06/21 09:56:37] [ info] [storage] version=1.2.0, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
2022-06-21 11:56:37[2022/06/21 09:56:37] [ info] [cmetrics] version=0.3.1

As said, reverting the `:stable` tag does not make sense but this bump could be counted as a major version change since increasing to this version breaks things with no other config changes.

**Comment by PettitWesley** [comment]
My perspective on this recent issue is that I fully understand the pain and its unfortunate that folks were impacted by the addition of config validation; furthermore, fixing config validation is a good thing, and it's not a backwards breaking change. The config options that all of you had in your configs never had any effect- so no features were removed. I think of the fluent bit config options as an API contract. Imagine you were calling a service API without adhering to its contract- failing to validate and fail those requests would be a bug IMO.

**Comment by james-skinner-deltatre** [comment]
well its absolutely a contract, but I don't agree it is a non-breaking change to that contract. If I made a similar change to a REST API without a major version bump and brought down production, that's on me, not the consumer of the API

even so it **is** a positive change to make and the real problem here is people (us included) not pinning the image version and expecting nothing to ever change under them.
